test/Microsoft.AspNetCore.Testing.Tests/TaskExtensionsTest.cs

```diff
@@ -0,0 +1,26 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Diagnostics;
+using System.Threading.Tasks;
+using Xunit;
+
+namespace Microsoft.AspNetCore.Testing
+{
+    public class TaskExtensionsTest
+    {
+        [Fact]
+        public async Task TimeoutAfterTest()
+        {
+            const double timeoutMilliseconds = 100;
+            var sw = new Stopwatch();
+            sw.Start();
+            await Assert.ThrowsAsync<TimeoutException>(async () => await Task.Run(() =>
+            {
+                while (true) { };
```

Comment:
This is going to burn a lot of CPU time for the lifetime of the process. I'd really suggest `Task.Delay` or something similar there.